src/wallet/rpcwallet.cpp

```diff
@@ -2553,6 +2561,19 @@ UniValue fundrawtransaction(const JSONRPCRequest& request)
             feeRate = CFeeRate(AmountFromValue(options["feeRate"]));
             overrideEstimatedFeerate = true;
         }
+
+        if (options.exists("subtractFeeFromAmount")) {
+            subtractFeeFromAmount = options["subtractFeeFromAmount"].get_array();
+            for (unsigned int idx = 0; idx < subtractFeeFromAmount.size(); idx++) {
+                string strAddress(subtractFeeFromAmount[idx].get_str());
+                CBitcoinAddress address(strAddress);
+                if (!address.IsValid())
+                    throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, string("Invalid Bitcoin address: ")+strAddress);
+                if (setSubtractFeeFromAmount.count(strAddress))
+                    throw JSONRPCError(RPC_INVALID_PARAMETER, string("Invalid parameter, duplicated address: ")+strAddress);
+                setSubtractFeeFromAmount.insert(strAddress);
```

Comment:
Should it throw an error if the given address is valid but is not among the outputs? (would have to check for this below, after retrieving the transaction).  It seems like in this case, the user is trying to pay the fee with one of the outputs but has made an error.